index.js

```diff
@@ -13,17 +16,33 @@ const Request = {
     c.uint.preencode(state, req.fork)
     c.fixed32.preencode(state, req.treeHash)
     m.manifest.preencode(state, req.manifest)
+
+    let flags = 0
+    if (req.blobs) flags |= 1
+    c.uint.preencode(state, flags)
+
+    if (req.blobs === null) return
+    c.uint.preencode(state, req.blobs.length)
+    c.fixed32.preencode(state, req.blobs.treeHash)
   },
   encode (state, req) {
     c.uint.encode(state, req.version)
     c.uint.encode(state, req.length)
     c.uint.encode(state, req.fork)
     c.fixed32.encode(state, req.treeHash)
     m.manifest.encode(state, req.manifest)
+
+    let flags = 0
+    if (req.blobs) flags |= 1
```

Comment:
Just esthetics at this point, so no big deal, but I'd prefer to put the blobs logic in one block

```
if (req.blobs) {
  flags |= 1
  c.uint.encode(state, req.blobs.length)
  c.fixed32.encode(state, req.blobs.treeHash)
}
```

(equivalent when decoding)

is cleaner